prefs/prefs.go

```diff
@@ -0,0 +1,56 @@
+package prefs
+
+import (
+	"pkg.re/essentialkaos/ek.v8/knf"
+	"pkg.re/essentialkaos/ek.v8/fmtc"
+	"os"
+	"pkg.re/essentialkaos/ek.v8/fsutil"
+)
+
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+const (
+	STORAGE_URL  = "storage:url"
+	STORAGE_PATH = "storage:path"
+)
+
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+type Preferences struct {
+	StorageURL string
+	StoragePath string
+}
+
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+func (p *Preferences) Validate() []error {
+	var errs []error
+
+	if !fsutil.IsExist(p.StoragePath) {
+		errs = append(errs, fmtc.Errorf("Storage path %s does not exist\n", p.StoragePath))
+	}
+
+	if !fsutil.IsWritable(p.StoragePath) {
+		errs = append(errs, fmtc.Errorf("Storage path %s is not writable\n", p.StoragePath))
+	}
+
+	return errs
+}
+
+func New(configPath string) (*Preferences) {
+	cnf, err := knf.Read(configPath)
+
+	if err != nil {
+		fmtc.Printf(err.Error())
```

Comment:
Try to avoid printing errors or warnings outside of cli. Return error text as error struct, and print it in CLI.